core/src/components/range/range-interface.ts

```diff
@@ -8,7 +8,15 @@ export interface RangeChangeEventDetail {
   value: RangeValue;
 }
 
+export interface RangeChangeStartEventDetail {
+  value: RangeValue;
+}
+
+export interface RangeChangeEndEventDetaill {
+  value: RangeValue;
+}
+
 export interface RangeCustomEvent extends CustomEvent {
-  detail: RangeChangeEventDetail;
+  detail: RangeChangeEventDetail | RangeChangeStartEventDetail | RangeChangeEndEventDetaill;
```

Comment:
Why are we making separate interfaces if the definitions are all the same?

Reply:
We'd agreed to use separate event details on the design doc, to avoid potential confusion and make them easier to update if we need new fields later. I'm open to changing it though if y'all don't think it'll look weird down the road 🤷‍♀️